run.sh

```diff
@@ -3,327 +3,247 @@
-    define('WP_DEBUG_DISPLAY', ${WP_DEBUG_DISPLAY:-true});
-
-core install:
-  url: ${AFTER_URL:-localhost:8080}
-  title: $DB_NAME
-  admin_user: $DB_USER
-  admin_password: $DB_PASS
-  admin_email: $ADMIN_EMAIL
-  skip-email: true
-EOF
-
-# WP-CLI bash completions
-# ------------------
-echo "
-. /etc/bash_completion.d/wp-cli
-" >> /root/.bashrc
-. /etc/bash_completion.d/wp-cli
-
-# Helpers
-# ---------------------
+main() {
+    h1 'Begin WordPress Installation'
+    init
+
+    h2 'Waiting for MySQL to initialize...'
+    while ! mysqladmin ping \
+        --host="$DB_HOST" \
+        --user="$DB_USER" \
+        --password="$DB_PASS" \
+        --silent >/dev/null; do
+        sleep 1
+    done
 
-RED='\033[0;31m'
-GREEN='\033[0;32m'
-ORANGE='\033[0;33m'
-PURPLE='\033[0;34m'
-CYAN='\033[0;36m'
-BOLD='\E[1m'
-NC='\033[0m'
+    h2 'Configuring WordPress'
+    configure
 
-h1() {
-  local len=$(($(tput cols)-1))
-  local input=$*
-  local size=$(((len - ${#input})/2))
+    h2 'Checking database'
+    check_database
 
-  for ((i = 0; i < len; i++)); do echo -ne "${PURPLE}${BOLD}="; done; echo ""
-  for ((i = 0; i < size; i++)); do echo -n " "; done; echo -e "${NC}${BOLD}$input"
-  for ((i = 0; i < len; i++)); do echo -ne "${PURPLE}${BOLD}="; done; echo -e "${NC}"
-}
+    if [[ "$MULTISITE" == 'true' ]]; then
+        h2 'Enabling Multisite'
+        wp --color core multisite-convert |& logger
+    fi
 
-h2() {
-  echo -e "${ORANGE}${BOLD}==>${NC}${BOLD} $*${NC}"
-}
+    h2 'Checking themes'
+    check_themes
 
+    h2 'Checking plugins'
+    check_plugins
 
-_colorize() {
-  local IN
-  local success="${GREEN}${BOLD}Success:${NC}"
-  local failed="${RED}${BOLD}Error:${NC}"
-  local warning="${CYAN}${BOLD}Warning:${NC}"
-  while read -r IN; do
-    IN="${IN/Success\:/$success}"
-    IN="${IN/Error\:/$failed}"
-    IN="${IN/Warning\:/$warning}"
-    echo -e "$IN"
-  done
-}
+    h2 'Finalizing'
+    if [[ "$MULTISITE" != 'true' ]]; then
+        wp --color rewrite structure "$PERMALINKS" --hard |& logger
+    fi
 
-_log_last_exit_colorize() {
-  if [ $? -eq 0 ]; then
-    echo "$1" |& _colorize
-  else
-    echo "$2" |& _colorize
-    exit 1
-  fi
-}
+    h1 'WordPress Configuration Complete!'
 
-_get_volumes() {
-  local volume_type="$1"
-  local filenames dirnames
-  local names=()
-
-  filenames=$(
-    find /app/wp-content/"$volume_type"/* -maxdepth 0 -type f ! -name 'index*' -print0 2>/dev/null |
-    xargs -0 -I {} basename {} .php
-  )
-  dirnames=$(
-    find /app/wp-content/"$volume_type"/* -maxdepth 0 -type d -print0 2>/dev/null |
-    xargs -0 basename -a 2>/dev/null
-  )
-  names=( $filenames $dirnames )
-
-  echo "${names[@]}"
-}
-
-_wp() {
-  wp --allow-root "$@"
+    sudo rm -f /var/run/apache2/apache2.pid
+    sudo apache2-foreground
 }
 
 # Config Functions
 # ---------------------
 
 init() {
-  local plugins themes i
-
-  PLUGINS="${PLUGINS/%,},"
-  THEMES="${THEMES/%,},"
-
-  if [[ -f /root/.dockercache ]]; then
```

Comment:
Is there a reason you removed this?

Reply:
Crap. Good catch.

Forgot that we needed to keep track of those beyond deciphering which of the existing themes/plugins are volumed.

I added the new `--skip-content` flag to the `wp core download` call on line 95 and I thought that would satisfy our needs.

The good news is that, even though we need to add this back in, it can be greatly simplified with `skip-content` enabled. 

I'll add that back in.